corehq/messaging/scheduling/tests/test_bulk_conditional_alerts.py

```diff
@@ -341,29 +348,114 @@ def test_upload(self, language_list_patch):
+                (self._get_rule(self.CUSTOM_IMMEDIATE_RULE).id, 'test', 'A Correction', 'Una Corrección'),
+            )),
+            ("not translated", (
+                (self._get_rule(self.CUSTOM_DAILY_RULE).id, 'test', 'Not Like This Train'),
+                (self._get_rule(self.CUSTOM_DAILY_RULE).id, 'test', 'Free Man in Nice'),
+            )),
+        )
+
+        test_cases = (self.CUSTOM_IMMEDIATE_RULE, self.CUSTOM_DAILY_RULE)
+        old_schedules = {test_case: self._get_rule(test_case).get_messaging_rule_schedule()
+                         for test_case in test_cases}
+
+        msgs = self._upload(data)
+
+        self.assertEqual(len(msgs), 2)
+        self.assertIn("Updated 1 rule(s) in 'translated' sheet", msgs)
+        self.assertIn("Updated 1 rule(s) in 'not translated' sheet", msgs)
+
+        custom_daily_rule = self._get_rule(self.CUSTOM_DAILY_RULE)
+        custom_daily_schedule = custom_daily_rule.get_messaging_rule_schedule()
+        self._assertTimedScheduleEventsEqual(custom_daily_schedule, old_schedules[self.CUSTOM_DAILY_RULE])
+        self._assertCustomContent(custom_daily_schedule, [
+            {'*': 'Not Like This Train'},
+            {'*': 'Free Man in Nice'},
+        ])
+
+        custom_immediate_rule = self._get_rule(self.CUSTOM_IMMEDIATE_RULE)
+        custom_immediate_schedule = custom_immediate_rule.get_messaging_rule_schedule()
+        self._assertAlertScheduleEventsEqual(custom_immediate_schedule, old_schedules[self.CUSTOM_IMMEDIATE_RULE])
+        self._assertCustomContent(custom_immediate_schedule, [{
+            'en': 'Plastic Bag',
+            'es': 'Bolsa de Plastico',
+        }, {
+            'en': 'A Correction',
+            'es': 'Una Corrección',
+        }])
+
+    @patch('corehq.messaging.scheduling.view_helpers.get_language_list')
+    def test_upload_custom_schedule_message_count_mismatch(self, language_list_patch):
+        language_list_patch.return_value = self.langs
+
+        data = (
+            ("translated", (
+                (self._get_rule(self.CUSTOM_IMMEDIATE_RULE).id, 'test', 'Plastic Bag', 'Bolsa de Plastico'),
+                (self._get_rule(self.CUSTOM_IMMEDIATE_RULE).id, 'test', 'Cloth Bag', 'Bolsa de Tela'),
+                (self._get_rule(self.CUSTOM_IMMEDIATE_RULE).id, 'test', 'Fishnet Bag', 'Bolsa de Red'),
+            )),
+            ("not translated", (
+                (self._get_rule(self.CUSTOM_DAILY_RULE).id, 'test', 'Like That Other Train'),
+            )),
+        )
+
+        msgs = self._upload(data)
+
+        self.assertEqual(len(msgs), 4)
+        self.assertIn("Updated 0 rule(s) in 'translated' sheet", msgs)
+        self.assertIn("Updated 0 rule(s) in 'not translated' sheet", msgs)
+        self._assertPatternIn(r"Could not update rule with id \d+ in 'translated' sheet: "
+                               "expected 2 .* but found 3.", msgs)
```

Comment:
don't think it matters here, but when you do continuation strings like this it's necessary to put the string prefix on each line:
```py
(r"Could not update rule with id \d+ in 'translated' sheet: "
 r"expected 2 .* but found 3.", msgs)
```

If there were a regex escape sequence (like `\b`) in the second line it would matter.